client/job.lua

```diff
@@ -268,43 +264,37 @@ end
 
 ---Creates a zone that lets players store and retrieve job vehicles
 ---@param vehicles AuthorizedVehicles
----@param vehiclePlatePrefix string
 ---@param coords vector4
-local function createGarage(vehicles, vehiclePlatePrefix, coords)
-
-    local function inVehicleZone()
-        if QBX.PlayerData.job.type == 'ems' and QBX.PlayerData.job.onduty then
-            lib.showTextUI(locale('text.veh_button'))
-            checkGarageAction(vehicles, vehiclePlatePrefix, coords)
-        else
-            checkVehicle = false
-            lib.hideTextUI()
-        end
-    end
-
-    local function outVehicleZone()
-        checkVehicle = false
-        lib.hideTextUI()
-    end
-
+local function createGarage(vehicles, coords)
     lib.zones.box({
         coords = coords.xyz,
         size = vec3(5, 5, 2),
         rotation = coords.w,
         debug = config.debugPoly,
-        inside = inVehicleZone,
-        onExit = outVehicleZone
+        inside = function()
+            if QBX.PlayerData.job.type == 'ems' and QBX.PlayerData.job.onduty then
+                lib.showTextUI(locale('text.veh_button'))
+                checkGarageAction(vehicles, coords)
+            else
+                checkVehicle = false
+                lib.hideTextUI()
+            end
+        end,
```

Comment:
```suggestion
        inside = function()
            if QBX.PlayerData.job.type == 'ems' and QBX.PlayerData.job.onduty then
                checkGarageAction(vehicles, coords)
            else
                checkVehicle = false
            end
        end,
        onEnter = function()
            if QBX.PlayerData.job.type == 'ems' and QBX.PlayerData.job.onduty then
                lib.showTextUI(locale('text.veh_button'))
            end
        end,
```

We shouldn't call showTextUI on every tick. 

Additionally depending on how many times we call the EMS check it might be worthwhile to put that into a reusable function for the resource or perhaps an extension to the lib